### DISCUSSION TITLE
Emergency Access Invite not working in HA integration

### DISCUSSION THREAD

**Discussion by therabbi99** [dani-garcia/vaultwarden#3183] [Q&A]
When I try to create an Trusted Emergency Access Contact the e-mail address I enter immediately shows up as "accepted" even though I know it hasn't been and that e-mail address hasn't been used before.  The user never got an invite e-mail.  I tried another e-mail address of my own and I never receive an invite either.

Everything else with this integration seems to be working great.  Anyone have an idea why this feature does not?

 Add-on version: 0.19.0
 You are running the latest version of this add-on.
 System: Home Assistant OS 9.4  (amd64 / qemux86-64)
 Home Assistant Core: 2023.1.7
 Home Assistant Supervisor: 2023.01.1

I don't see any errors in the log.

[2023-01-27 22:25:48.530][request][INFO] POST /api/emergency-access/invite
[2023-01-27 22:25:48.532][response][INFO] (send_invite) POST /api/emergency-access/invite => 200 OK

Thanks in advance!

**Comment by BlackDex** [comment]
Vaultwarden isn't build with HA in mind, but it could with some points to keep in mind. Read the following post https://github.com/dani-garcia/vaultwarden/discussions/3143#discussioncomment-4685904

[a comment by an automated account is not shown]

**Comment by BlackDex** [comment]
Then it's something on his mail server or somewhere in between.
If Vaultwarden doesn't report an error, it's difficult to fix.

**Reply by BlackDex** [reply]
Same issue with NPM, could be NodeJS Package Manager or Nginx Proxy Manger.

[a comment by an automated account is not shown]

**Comment by stefan0xC** [comment]
If there are no errors my first suspicion would be that you have not [configured SMTP](https://github.com/dani-garcia/vaultwarden/wiki/SMTP-Configuration) at all, so there won't be any mail sent.

@BlackDex We should probably handle the accepting of emergency access invites on registration like we do with organization invites if mail is not enabled?
https://github.com/dani-garcia/vaultwarden/blob/897bdf8343148ee4e4f1a937cbc501ad3f743b4a/src/api/core/emergency_access.rs#L260-L262

**Reply by BlackDex** [reply]
Well, no actually.
That could be a security risk.

Then the account owner doesn't receive a message that someone requested access.

This feature should not work if smtp is disabled at all.

**Reply by stefan0xC** [reply]
They would still [have to confirm](https://bitwarden.com/help/emergency-access/#tab-confirm-2GQ11N6qIbcr9F85DD8gjl), wouldn't they? (I mean initially, if they want to grant someone access, not when they initiate access.)
Disabling the feature due to security concerns would be fine to me too. As it is, it's a bit broken (because the invited user will be accepted before the account exists) but it would still work.

**Reply by BlackDex** [reply]
Only confirm that the user is allowed to be a contact. After that if someone requests access, the account owner doesn't know the the emergency contact person request access at all. If that is a view only, then that person can just see the passwords without the account owner ever knowing it ever happened.

So, this feature should only work with valid SMTP mail settings, and should fail if for some reason the mail server is not reachable.
Of course, we can't verify if the mail reached the user or not, but it should at least fail if it wasn't able to send out the email.

**Reply by stefan0xC** [reply]
The question would be if that is a valid security concern or something users allowing emergency access on a mail disabled system would have to take into account, when they grant someone access. Not sure if we should prevent this from ever happening or the best way to go about it. I mean, I think the rationale behind this feature is to give someone else access to your passwords. If there's no trust, then they should not give someone the emergency access in the first place (or they should at least set a higher wait time and check regularly). Preventing the misuse of this trust is not something we can really prevent. (Like you said, even when mail is enabled, it's not a guarantee that a person really receives a successfully sent mail.) And I don't think that there's a best way to go about it. (Like I said I can live with disabling this feature but it would not be my first choice of action.) If we decide to keep the feature we should probably make `emergency_access_allowed` dependent on whether `mail_enabled` is true, so it would have to be an explicit choice to enable the feature if mail is not configured.
